ui/src/wallet.ts

```diff
@@ -200,12 +235,54 @@ class WalletConnectSigner extends ethers.Signer {
     throw new Error("not implemented");
   }
 
+  async sendTransaction(
+    transaction: Deferrable<TransactionRequest>
+  ): Promise<TransactionResponse> {
+    if (this._environment === ethereum.Environment.Local) {
+      return super.sendTransaction(transaction);
+    }
```

Comment:
So this is essentially configurable behaviour through arguments, which I am not usually a fan of. The alternative would be to have a dedicated implementation of `ethers.Signer` for the `Local` and another for the `Ropsten` environments, stripped away of configurable behaviour.

I didn't go that route since we only need the check once. Let me know if you would like us to revisit this :hand:

Reply:
I am not sure how to phrase it though in a way that adds more information than what the code already provides. Do you have a recommendation?